pkg/core/resources/manager/manager.go

```diff
@@ -123,7 +123,7 @@ func NewUpsertOpts(fs ...UpsertFunc) UpsertOpts {
 	return opts
 }
 
-func Upsert(manager ResourceManager, key model.ResourceKey, resource model.Resource, fn func(resource model.Resource), fs ...UpsertFunc) error {
```

Comment:
Is there a way to give a better name to the `fn` parameter or to type the `func`? It's tricky to give a good name because it often relies on side effect but maybe `shouldUpdate(res)`